src/config/config.rs

```diff
@@ -139,17 +153,178 @@ pub struct CSIConfig {
     pub worker_port: u16,
 }
 
+impl Config {
+    /// Load config from command line, config file and environment variables
+    ///
+    /// Priority: command line > config file > environment variables
+    #[inline]
+    pub fn load_from_args(arg_conf: Self) -> Result<Self, DatenLordError> {
+        let mut builder: serfig::Builder<Self> = serfig::Builder::default();
+
+        // Load from config file
+        if let Some(config_file) = arg_conf
+            .config_file
+            .clone()
+            .or_else(|| std::env::var("CONFIG_FILE").ok())
+        {
+            builder = builder.collect(from_file(Toml, &config_file));
+        }
+
+        // Load from environment
+        builder = builder.collect(from_env());
+
+        // Load from arguments
+        builder = builder.collect(from_self(arg_conf));
+
+        // Build the config with multi sources
+        let conf = builder.build()?;
+
+        // Check valid
+        if conf.check_valid() {
+            Ok(conf)
+        } else {
+            Err(DatenLordError::ArgumentInvalid {
+                context: vec!["Current command line arguments are not valid.".to_owned()],
```

Comment:
As you've checked the configuration's valication first, you'd better tell the user which parameter is not right, which is a helpful fix hint.

Reply:
Thanks, I will change it.